lib/api/root.rb

```diff
@@ -59,19 +63,32 @@ def call(object, _env)
 
     parser :json, Parser.new
 
+    use OpenProject::Authentication::Manager
+
     helpers do
       def current_user
-        return User.current if running_in_test_env?
-        user_id = env['rack.session']['user_id']
-        User.current = user_id ? User.find(user_id) : User.anonymous
+        User.current
+      end
+
+      def warden
+        env['warden']
       end
 
       def authenticate
-        if Setting.login_required? && (current_user.nil? || current_user.anonymous?)
+        warden.authenticate! scope: API_V3
+
+        User.current = warden.user scope: API_V3
```

Comment:
That means you moved the side-effect "sets `User.current`" from `current_user` to `authenticate`, right?

This sounds reasonable, but you should update this: https://github.com/machisuji/openproject/blob/846e331c784fcc2a9ae1ad42cae3adf65a273454/lib/api/root.rb#L165-L173